packages/mattermost-redux/src/selectors/entities/channels.ts

```diff
@@ -89,8 +92,28 @@ export function getChannelsInTeam(state: GlobalState): RelationOneToMany<Team, C
     return state.entities.channels.channelsInTeam;
 }
 
-export function getChannelsInPolicy(state: GlobalState): IDMappedObjects<Channel> {
-    return state.entities.channels.channelsInPolicy;
+export function getChannelsInPolicy() {
+    return (createSelector(
+        getAllChannels,
+        (state: GlobalState, props: {policyId: string}) => getDataRetentionCustomPolicy(state, props.policyId),
+        (getAllChannels, policy) => {
+            if (!policy) {
+                return [];
+            }
+
+            const policyChannels: Channel[] = [];
+
+            Object.entries(getAllChannels).forEach((item: [string, Channel]) => {
```

Comment:
```suggestion
            Object.entries(getAllChannels).forEach((channel: [string, Channel]) => {
```

Reply:
`item` isn't a channel, it's an array containing a channel. channelEntry might be a better name?